pkg/apis/configuration/v1/kongconsumer_types.go

```diff
@@ -45,6 +45,9 @@ type KongConsumer struct {
 	// Credentials are references to secrets containing a credential to be
 	// provisioned in Kong.
 	Credentials []string `json:"credentials,omitempty"`
+	// ConsumerGroups are references to Consumer Groups (that Consumer wants to be part of)
+	// provisioned in Kong.
```

Comment:
```suggestion
	// ConsumerGroups contains a list of consumer groups this consumer is a member of.
```
Kong docs do not capitalize their resource names by convention, e.g. https://docs.konghq.com/gateway/3.3.x/admin-api/consumer-groups/reference/

The phrasing on Credentials is iffy in its own right, but in either case we will provision consumer groups directly from a KongConsumerGroup, whether it's attached to anything or not. Credentials we only provision if they're listed on a KongConsumer.

Accepting this requires re-running `make generate` after.